main/src/library/pkger/pkger.flix

```diff
@@ -0,0 +1,43 @@
+pub enum SemVerComponent {
+    case Integer(Int),
+    case Wildcard,
+    case NotPresent
+}
+
+pub opaque type SemVer = { major: SemVerComponent
+                         , minor: SemVerComponent
+                         , patch: SemVerComponent
+                         }
+
+// Temporary
+pub opaque type SpecificSemVer = SemVer
+
+pub opaque type Package = { name: String
+                          , version: SpecificSemVer
+                          , authors: List[String]
+                          , description: String
+                          // This could be an enum. Would only support predefined licenses, or require an `Other` variant
+                          , license: Option[String]
+                          // Mapping package names to the semver pattern
+                          , depends: List[(String, SemVer)]
+                          , homepage: Option[String] // Url?
+                          }
+
+pub opaque type LockedPackage = { name: String
+                                // All transitive dependencies, their specific version, and where they were retrieved from.
+                                , transitive: List[(String, List[SpecificSemVer])]
+                                }
+
+// Add a new package to the package. Handle superseding old versions
+pub def addDependency(self: Package, package: String, version: SemVer): Package = ???
+
+// Remove a package from the list of dependencies
+pub def removeDependency(self: Package, package: String): Package = ???
+
+// Resolve dependency tree and create a list of all dependencies and their transitive dependencies
+pub def lockPackage(self: Package): List[LockedPackage] = ???
+
+// Create a new package. Everything but the name of the package can be populated with sensible defaults from the environment
+pub def initPackage(name: String): Array[String] & Impure = [name]
```

Comment:
I think this is a good place to start. You can port much of the functionality from Packager.scala to Flix to setup an initial project.